pageserver/src/walingest.rs

```diff
@@ -1241,7 +1288,7 @@ impl WalIngest {
                     if is_commit {
                         NeonWalRecord::ClogSetCommitted {
                             xids: page_xids,
-                            timestamp: parsed.xact_time,
+                            timestamp: warn_ingest_delay(parsed.xact_time),
```

Comment:
If the warning is only controlled by `parsed`, why are we doing the check down here in the loop over `subxacts`?  It seems like we should check it one time before the `for subxact...` loop, rather than inside.